### ISSUE TITLE
Problem with _data_member_location_extra

### ISSUE THREAD

**Issue by bseifers** [eliben/pyelftools#97]
I got a python error when running readelf.py on a basic demo C file. I was able to debug it and put a kludge in your code to work around it, but you probably know a better way to make a more permanent fix.

To recreate the problem, create a hello.out file using the source files I have included (hello.c/h):
gcc -g -o hello.out hello.c
(or just use the .out file I have submitted)

Then run 
readelf.py --debug-dump=info hello.out
which gives the following python error: 

Traceback (most recent call last):
  File "readelf.py", line 1177, in <module>
    main()
  File "readelf.py", line 1156, in main
    readelf.display_debug_dump(options.debug_dump_what)
  File "readelf.py", line 658, in display_debug_dump
    self._dump_debug_info()
  File "readelf.py", line 880, in _dump_debug_info
    attr, die, section_offset)))
  File "/home/bseifers/.local/lib/python2.6/site-packages/elftools/dwarf/descriptions.py", line 38, in describe_attr_value
    extra_info = extra_info_func(attr, die, section_offset)
  File "/home/bseifers/.local/lib/python2.6/site-packages/elftools/dwarf/descriptions.py", line 442, in _data_member_location_extra
    return describe_DWARF_expr(attr.value, die.cu.structs)
  File "/home/bseifers/.local/lib/python2.6/site-packages/elftools/dwarf/descriptions.py", line 149, in describe_DWARF_expr
    dwarf_expr_dumper.process_expr(expr)
  File "/home/bseifers/.local/lib/python2.6/site-packages/elftools/dwarf/dwarf_expr.py", line 119, in process_expr
    self.stream = BytesIO(bytelist2string(expr))
  File "/home/bseifers/.local/lib/python2.6/site-packages/elftools/common/utils.py", line 19, in bytelist2string
    return b''.join(int2byte(b) for b in bytelist)
TypeError: 'int' object is not iterable

I was able to fix the problem by modifying elftools/dwarf/descriptions.py as follows

def _data_member_location_extra(attr, die, section_offset):
    # According to section 5.5.6 of the DWARF spec v4, a data member location
    # can be an integer offset, or a location description.
    #
    if attr.form in ('DW_FORM_data1', 'DW_FORM_data2',
                     'DW_FORM_data4', 'DW_FORM_data8'):
        return ''  # No extra description needed
# Start bseifers kludge

```
elif attr.form in ('DW_FORM_sdata'):
return str(attr.value)
```
# End bseifers kludge

```
else:
    return describe_DWARF_expr(attr.value, die.cu.structs)
```

I'm having trouble attaching the C files, so I'll include the source text here:

hello.c:
# include <stdio.h>
# include "hello.h"

const int GLOBAL_CONST;

int tryGlobal;
struct def hiLo;

int main()
{
    int abc;
    printf("Hello World\n");
    return 0;
}

hello.h:
# ifndef hello_h
# define hello_h

struct def
{
    int ijk;
    char c;
    long long lint;
    float mno;
    int bit1 : 1;
    int bit3 : 3;
    int bit2 : 2;
    int bit4 : 4;
//};
}**attribute**((**packed**));
# endif

file:///home/bseifers/hello.c
file:///home/bseifers/hello.h
file:///home/bseifers/hello.out

**Comment by eliben** [comment]
Please create a patch and open a Pull Request. Add tests. Then it will be much easier to discuss the code changes and see them in diff context with the existing code.

**Comment by bseifers** [comment]
OK, after much thrashing, I managed to create a pull request, including a test file that demonstrates the problem and fix. As I say in the comments, there is probably a better way, but I don’t know enough about your tool to do any better. What I have works for my case, which is all I care about. Hope this helps.

Brad
408-742-4764

**Comment by eliben** [comment]
Thanks. I'll take a look at the PR - hopefully this week (I'm swamped).

**Comment by bseifers** [comment]
No rush, whenever you get the chance…

Brad
408-742-4764

**Comment by eliben** [comment]
Fixed by PR #98
